### ISSUE TITLE
[bug] could not compile `tauri-cli` (bin "cargo-tauri")

### ISSUE THREAD

**Issue by asthetik** [tauri-apps/tauri#7380]
### Describe the bug

could not compile `tauri-cli` (bin "cargo-tauri")

<img width="1048" alt="image" src="https://github.com/tauri-apps/tauri/assets/121742785/73145a29-4937-4001-8aac-02c80e8a6f7d">

error: could not compile `tauri-cli` (bin "cargo-tauri")

Caused by:
  process didn't exit successfully: `rustc --crate-name cargo_tauri --edition=2021 /home/codespace/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tauri-cli-1.4.0/src/main.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --diagnostic-width=237 --crate-type bin --emit=dep-info,link -C opt-level=3 -C lto --cfg 'feature="default"' --cfg 'feature="rustls"' -C metadata=baa9ccd79b2ac1f2 -C extra-filename=-baa9ccd79b2ac1f2 --out-dir /tmp/cargo-installVMQGpN/release/deps -L dependency=/tmp/cargo-installVMQGpN/release/deps --extern anyhow=/tmp/cargo-installVMQGpN/release/deps/libanyhow-e11a1eecba5167b0.rlib --extern axum=/tmp/cargo-installVMQGpN/release/deps/libaxum-9ae59477901b2c73.rlib --extern base64=/tmp/cargo-installVMQGpN/release/deps/libbase64-378915bcc77ad3d9.rlib --extern clap=/tmp/cargo-installVMQGpN/release/deps/libclap-a2bb5b795041fa3c.rlib --extern clap_complete=/tmp/cargo-installVMQGpN/release/deps/libclap_complete-a294003f74355650.rlib --extern colored=/tmp/cargo-installVMQGpN/release/deps/libcolored-04b9430203bb978e.rlib --extern common_path=/tmp/cargo-installVMQGpN/release/deps/libcommon_path-3da2cbd727df5788.rlib --extern ctrlc=/tmp/cargo-installVMQGpN/release/deps/libctrlc-3ccceb9df471321f.rlib --extern dialoguer=/tmp/cargo-installVMQGpN/release/deps/libdialoguer-f9306d435bba829e.rlib --extern env_logger=/tmp/cargo-installVMQGpN/release/deps/libenv_logger-ed0d91213fa5fd5e.rlib --extern handlebars=/tmp/cargo-installVMQGpN/release/deps/libhandlebars-b2ab1b68a04afdf5.rlib --extern heck=/tmp/cargo-installVMQGpN/release/deps/libheck-b1c7579fe9262c01.rlib --extern html5ever=/tmp/cargo-installVMQGpN/release/deps/libhtml5ever-b19c34b1eec4ff32.rlib --extern ignore=/tmp/cargo-installVMQGpN/release/deps/libignore-455fbbab89178301.rlib --extern image=/tmp/cargo-installVMQGpN/release/deps/libimage-457147163749a3d7.rlib --extern include_dir=/tmp/cargo-installVMQGpN/release/deps/libinclude_dir-2a56adbfe6c0dc0c.rlib --extern itertools=/tmp/cargo-installVMQGpN/release/deps/libitertools-1058a13b2fa9ca99.rlib --extern json_patch=/tmp/cargo-installVMQGpN/release/deps/libjson_patch-93e25208f41effa1.rlib --extern jsonschema=/tmp/cargo-installVMQGpN/release/deps/libjsonschema-354ce6e3dde78d16.rlib --extern kuchiki=/tmp/cargo-installVMQGpN/release/deps/libkuchiki-c533e2ca66522410.rlib --extern libc=/tmp/cargo-installVMQGpN/release/deps/liblibc-06b5fbe03d02e1c8.rlib --extern log=/tmp/cargo-installVMQGpN/release/deps/liblog-316c4764838fce9a.rlib --extern minisign=/tmp/cargo-installVMQGpN/release/deps/libminisign-032439fb9ea8c414.rlib --extern notify=/tmp/cargo-installVMQGpN/release/deps/libnotify-0347fc4f3a82eec5.rlib --extern notify_debouncer_mini=/tmp/cargo-installVMQGpN/release/deps/libnotify_debouncer_mini-21cb8b65268c0012.rlib --extern once_cell=/tmp/cargo-installVMQGpN/release/deps/libonce_cell-9a7561dee1dbe31c.rlib --extern os_info=/tmp/cargo-installVMQGpN/release/deps/libos_info-d7da7cfb4a5126cf.rlib --extern os_pipe=/tmp/cargo-installVMQGpN/release/deps/libos_pipe-36aa9e08bcfd64cd.rlib --extern regex=/tmp/cargo-installVMQGpN/release/deps/libregex-6bc90dcd49d21d01.rlib --extern semver=/tmp/cargo-installVMQGpN/release/deps/libsemver-0613945e2f225608.rlib --extern serde=/tmp/cargo-installVMQGpN/release/deps/libserde-50653216282fdc62.rlib --extern serde_value=/tmp/cargo-installVMQGpN/release/deps/libserde_value-994cb9a8f19b6216.rlib --extern serde_json=/tmp/cargo-installVMQGpN/release/deps/libserde_json-829f454754ad1bcb.rlib --extern shared_child=/tmp/cargo-installVMQGpN/release/deps/libshared_child-20813684b1ef10db.rlib --extern tauri_bundler=/tmp/cargo-installVMQGpN/release/deps/libtauri_bundler-e50db6dbe26107d8.rlib --extern tauri_cli=/tmp/cargo-installVMQGpN/release/deps/libtauri_cli-64ce9d60422f9faf.rlib --extern icns=/tmp/cargo-installVMQGpN/release/deps/libtauri_icns-71d187a67fdcb3b2.rlib --extern tauri_utils=/tmp/cargo-installVMQGpN/release/deps/libtauri_utils-27262bd10a5fc41e.rlib --extern tokio=/tmp/cargo-installVMQGpN/release/deps/libtokio-b6c3a7dc56368bd8.rlib --extern toml=/tmp/cargo-installVMQGpN/release/deps/libtoml-621c3e5f4455aa3d.rlib --extern toml_edit=/tmp/cargo-installVMQGpN/release/deps/libtoml_edit-82dc19acd1e7d493.rlib --extern unicode_width=/tmp/cargo-installVMQGpN/release/deps/libunicode_width-3b2c52aa848abb50.rlib --extern ureq=/tmp/cargo-installVMQGpN/release/deps/libureq-934412d040e0e53e.rlib --extern url=/tmp/cargo-installVMQGpN/release/deps/liburl-643ea8200e234acb.rlib --extern zeroize=/tmp/cargo-installVMQGpN/release/deps/libzeroize-4f8f8d0122019534.rlib --cap-lints allow -L native=/tmp/cargo-installVMQGpN/release/build/ring-eac3c16e87146446/out -L native=/tmp/cargo-installVMQGpN/release/build/bzip2-sys-cac1298e3f153909/out/lib -L native=/tmp/cargo-installVMQGpN/release/build/zstd-sys-b399b13a7158e8ff/out` (signal: 15, SIGTERM: termination signal)
error: failed to compile `tauri-cli v1.4.0`, intermediate artifacts can be found at `/tmp/cargo-installVMQGpN`

### Reproduction

```
/workspaces $ cargo install create-tauri-app --locked
/workspaces $ cargo create-tauri-app
✔ Project name · tauri-app
✔ Choose which language to use for your frontend · Rust - (cargo)
✔ Choose your UI template · Yew - (https://yew.rs/)

Template created!

Your system is missing dependencies (or they do not exist in $PATH):
╭────────────────────┬──────────────────────────────────────────────────────────────────────────────────╮
│ Tauri CLI          │ Run `cargo install tauri-cli`                                                    │
├────────────────────┼──────────────────────────────────────────────────────────────────────────────────┤
│ Trunk              │ Run `cargo install trunk`                                                        │
├────────────────────┼──────────────────────────────────────────────────────────────────────────────────┤
│ wasm32 target      │ Run `rustup target add wasm32-unknown-unknown`                                   │
├────────────────────┼──────────────────────────────────────────────────────────────────────────────────┤
│ webkit2gtk & rsvg2 │ Visit https://tauri.app/v1/guides/getting-started/prerequisites#setting-up-linux │
╰────────────────────┴──────────────────────────────────────────────────────────────────────────────────╯

Make sure you have installed the prerequisites for your OS: https://tauri.app/v1/guides/getting-started/prerequisites, then run:
  cd tauri-app
  cargo tauri dev

/workspaces $ cargo install tauri-cli
    Updating crates.io index
  Downloaded tauri-cli v1.4.0
  Downloaded 1 crate (1.2 MB) in 0.30s
  Installing tauri-cli v1.4.0
    Updating crates.io index
# ...
```

when i run the commend`cargo install tauri-cli`, it causes the error.

### Platform and versions

```text
/workspaces $ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.6 LTS
Release:        20.04
Codename:       focal
/workspaces $ rustc --version
rustc 1.70.0 (90c541806 2023-05-31)
```

**Comment by FabianLars** [comment]
Thanks for the report! Can you try this command? `cargo install tauri-cli --locked`.

If that still doesn't work, please follow https://tauri.app/v1/guides/getting-started/prerequisites#setting-up-linux first and then try again.

**Comment by asthetik** [comment]
I re-downloaded these necessary dependencies， still the same error.

### see the shell history
```
@asthetik ➜ /workspaces $ history
    1  sudo apt update
    2  sudo apt install libwebkit2gtk-4.0-dev     build-essential     curl     wget     file     libssl-dev     libgtk-3-dev     libayatana-appindicator3-dev     librsvg2-dev \
    3  curl --proto '=https' --tlsv1.2 https://sh.rustup.rs -sSf | sh
    4  source "$HOME/.cargo/env"
    5  rustc --version --verbose
    6  cargo install create-tauri-app --locked
    7  cargo install tauri-cli
    8  history
    9  cargo install tauri-cli --locked
   10  history
```
### detailed shell logs
for your information.

[shell logs.txt](https://github.com/tauri-apps/tauri/files/11985130/shell.logs.txt)

**Comment by FabianLars** [comment]
Did you also try it with the `--locked` flag as mentioned above?
```
cargo install tauri-cli --locked
```
This will make sure it pulls exactly the dependency versions we used when publishing the cli

**Comment by asthetik** [comment]
yes, i also try it with the --locked flag, still failed. it's displayed in my shell history.

**Comment by FabianLars** [comment]
ah sorry, missed that part. btw, until we figure out the issue you can download the cli from [here](https://github.com/tauri-apps/tauri/releases/tag/tauri-cli-v1.4.0) and move it into `~/.cargo/bin/` (effectively what cargo install does)

**Comment by asthetik** [comment]
> ah sorry, missed that part. btw, until we figure out the issue you can download the cli from [here](https://github.com/tauri-apps/tauri/releases/tag/tauri-cli-v1.4.0) and move it into `~/.cargo/bin/` (effectively what cargo install does)

thank you! Let me try again

**Comment by MrGcGamer** [comment]
I've got the same issue.
Seems to be an env problem of some sort with GitHub Codespaces.

**Comment by gralpli** [comment]
@asthetik I had exactly the same problem and the reason for me was that I had too little memory (I ran it in a VM that had only 2 GiB). Could that have been the reason for you as well?

**Comment by asthetik** [comment]
@gralpli yes, i used 8GB of memory and the compilation worked normally. Thank you!

**Comment by ahpb21** [comment]
I had the same problem, it was caused by an incorrect environment for rust, you need to change your rust environment to MSVC so it will run.

**Comment by msrd0** [comment]
Needing 8GB of ram to compile a single thing is a huge problem. I just ran into this issue as well. Adding more memory is not an option. Please try to reduce the amount of memory necessary to compile `tauri-cli`.
